app/assets/javascripts/admin/index_utils/services/paged_fetcher.js.coffee

```diff
@@ -2,19 +2,26 @@ angular.module("admin.indexUtils").factory "PagedFetcher", (dataFetcher) ->
   new class PagedFetcher
     # Given a URL like http://example.com/foo?page=::page::&per_page=20
     # And the response includes an attribute pages with the number of pages to fetch
-    # Fetch each page async, and call the processData callback with the resulting data
-    fetch: (url, processData, onLastPageComplete) ->
-      dataFetcher(@urlForPage(url, 1)).then (data) =>
-        processData data
+    # Fetch each page async, and call the pageCallback callback with the resulting data
+    # Developer note: this class should not be re-used!
```

Comment:
:joy:

Reply:
It felt  like it would have been irresponsible to touch this file without leaving this comment :see_no_evil:

Reply:
This `PagedFetcher` is only used in one place, in inventory. When we either switch to the network model or spend time implementing pagination here we can get rid of this class, but until then I don't think it should be used elsewhere.